libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs

```diff
@@ -1194,6 +1195,29 @@ private string GetBotAppId(ITurnContext turnContext)
             return appId;
         }
 
+        /// <summary>
+        /// This method creates a default ConnectorClient for the bot AppId and also registers the service URL as a trusted URL.
+        /// </summary>
+        /// <remarks>
+        /// When a parent bot is deployed to multiple instances the cache AppIds, ConnectClients and Trusted URL are not initialized
+        /// if the server instance hasn't been hit by a request from the channel.
+        /// This code ensures that the required objects are created.
+        /// </remarks>
+        private async Task EnsureChannelConnectorClientIsCreatedAsync(string serviceUrl, ClaimsIdentity claimsIdentity, CancellationToken cancellationToken)
+        {
+            // Ensure we have a default ConnectorClient and MSAppCredentials instance for the audience.
+            var audience = claimsIdentity.Claims.FirstOrDefault(claim => claim.Type == AuthenticationConstants.AudienceClaim)?.Value;
+            if (string.IsNullOrWhiteSpace(audience) || !AuthenticationConstants.ToBotFromChannelTokenIssuer.Equals(audience, StringComparison.InvariantCulture))
```

Comment:
I wonder if it would be more appropriate to use `StringComparison.InvariantCultureIgnoreCase`

Reply:
Changed it to InvariantCultureIgnoreCase